### PR TITLE
nextcloud: allow to configure proxy options via env

### PR DESCRIPTION
**Pull request by szaimen**

### CODE REVIEW

**Review by st3iny — APPROVED** [review]
Looks good!

I just noticed the following paragraph in our documentation:

> Note: Guzzle (the http library used by Nextcloud) is reading the environment variables HTTP_PROXY (only for cli request), HTTPS_PROXY, and NO_PROXY by default.

Syncing explicitly is a good idea though.

File: Containers/nextcloud/config/proxy.config.php [hunk 1]
```diff
@@ -0,0 +1,13 @@
+<?php
+if (getenv('HTTP_PROXY') !== false) {
+    $CONFIG['proxy'] = getenv('HTTP_PROXY');
+}
+if (getenv('HTTPS_PROXY') !== false) {
+    $CONFIG['proxy'] = getenv('HTTPS_PROXY');
+}
```
Comment (st3iny) [inline Containers/nextcloud/config/proxy.config.php:7]: Can't compare against `false` here.

Problem: When setting `HTTP_PROXY` to some value but not setting `HTTPS_PROXY` (in the Helm chart), then `$CONFIG['proxy']` will be an empty string (`''`).

This happens because values are **always** passed down to containers, even if they are not set (which means they are really set to empty strings). Then, the second value (`getenv('HTTPS_PROXY')`) is an empty string instead of false and thus the proxy config will be overwritten.

I'm sorry for the late test :see_no_evil: I think this is the first time we encounter this edge case of overwriting the same config value within two if blocks.

Comment (st3iny) [reply Containers/nextcloud/config/proxy.config.php:7]:
```php
// Executed inside the mgmt container
php > var_dump(getenv('HTTP_PROXY'));
string(19) "100.86.114.111:8080"
php > var_dump(getenv('HTTPS_PROXY'));
string(0) ""
```

Comment (szaimen) [reply Containers/nextcloud/config/proxy.config.php:7]: Thanks Richard! I did not know that there is a difference between empty string and not existing variable...
I opened https://github.com/nextcloud/all-in-one/pull/5117 to fix this :)
